salt/states/dockerng.py

```diff
@@ -330,6 +331,14 @@ def _compare(actual, create_kwargs, runtime_kwargs):
                                        'new': desired_hosts}})
                     continue
 
+            elif item == 'volumes':
+                difference = set(data).difference(set(actual_data))
+                desired_data = sorted(data)
+                if difference:
+                    ret.update({item: {'old': sorted(actual_data),
+                                       'new': desired_data}})
+                continue
+
             elif isinstance(data, list):
```

Comment:
Similar to the other diff, this is calculating the difference. The exposed volumes are just a list, so comparing sorted lists is sufficient, as is explained in the long comment just below.